src/DataTable.php

```diff
@@ -800,6 +851,52 @@ public function buildSearch(): Builder
             }
         }
 
+        // include selected relationships
+        $tmpWith = [];
+        $with = [];
+        $loadedModels = [];
+        $baseModelInfo = ModelInfo::forModel($this->model);
+        $modelQuery = new $this->model;
+        foreach ($this->enabledCols as $enabledCol) {
+            $foreignKeys = [];
+            if (str_contains($enabledCol, '.')) {
+                $explodedCol = explode('.', $enabledCol);
+                $attribute = Arr::pull($explodedCol, count($explodedCol) - 1);
+                $path = implode('.', array_map(fn ($relation) => Str::camel($relation), $explodedCol));
+                $relation = $baseModelInfo->relation(Str::camel($path));
+
+                $relatedModel = $relation->related;
+                $loadedModels[$path] = $relatedModel;
+
+                if (! ModelInfo::forModel($relatedModel)->attribute($attribute)->virtual) {
+                    $relationInstance = $modelQuery->{Str::camel($path)}();
+                    if (! method_exists($relationInstance, 'getOwnerKeyName')
+                        && ! method_exists($relationInstance, 'getForeignKeyName')
+                    ) {
+                        continue;
+                    }
+
+                    $foreignKeys[] = method_exists($relationInstance, 'getOwnerKeyName')
+                        ? $relationInstance->getOwnerKeyName()
+                        : $relationInstance->getForeignKeyName();
+                    $tmpWith[$path] = array_unique(
+                        array_merge($foreignKeys, $tmpWith[$path] ?? [], [$attribute])
+                    );
+                } else {
+                    $with[] = $path;
```

Comment:
What if multiple columns from same related table get included? Shouldn't it be e.g. users:id,name instead of [users:id, users:name]